extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -148,11 +163,11 @@ class AmpFlyingCarpet extends AMP.BaseElement {
    */
   visibileChildren_(nodes) {
     return nodes.filter(node => {
-      if (node.nodeType === /* Element */1) {
+      if (node.nodeType === /* Element */ 1) {
         return true;
       }
 
-      if (node.nodeType === /* Text */3) {
+      if (node.nodeType === /* Text */ 3) {
         // Is there a non-whitespace character?
         return /\S/.test(node.textContent);
```

Comment:
lol, problem with regex is that its CPU intensive. Both of your tests, when run on actual phone, are still 30%-50% slower (sure on desktop difference in negligible)